src/repeatn.rs

```diff
@@ -44,6 +44,20 @@ where
     fn size_hint(&self) -> (usize, Option<usize>) {
         (self.n, Some(self.n))
     }
+
+    fn fold<B, F>(self, mut init: B, mut f: F) -> B
+    where
+        F: FnMut(B, Self::Item) -> B,
+    {
+        match self {
+            Self { elt: Some(elt), n } => {
```

Comment:
Here, we take `elt` as reference if there's still work to do, whereas `fn next` uses `self.n > 1`. Should we canonicalize?

Reply:
I'm not sure there is a real benefit to win here. Feel free to investigate.